services/server/src/handlers/topic.ts

```diff
@@ -170,12 +178,54 @@ async function addUserTopics(req: Request, res: Response, next: NextFunction) {
   res.sendStatus(200);
 }
 
+async function replaceUserTopics(req: Request, res: Response, next: NextFunction) {
+  const user = req.user;
+  const allowed = isStudent(user) || isTutor(user) || isTutorManager(user);
+  if (!allowed) return next(forbidden());
+
+  const { removeTopics, addTopics }: ITopic.ReplaceUserTopicsApiPayload = 
+    replaceUserTopicsPayload.parse(req.body);
+
+  if (isEmpty(removeTopics) && isEmpty(addTopics)) return next(empty());
+
+  // ensure that all oldTopics exists for the user
+  const inDBTopics = await topics.findUserTopics({ users: [user.id] });
+  const inDBTopicIds = inDBTopics.map(topic => topic.id);
+  for (const topicId of removeTopics) {
+    if (!inDBTopicIds.includes(topicId))
+      return next(notfound.topic());
+  }
+  
+  // verify that all topics do exist in the database
+  const isExists = await topics.isExistsBatch(addTopics);
+  if (Object.values(isExists).includes(false))
+    return next(notfound.topic());
+
+  // ensure that user topics will not exceed the max num after insertion
+  if ((inDBTopicIds.length + addTopics.length - removeTopics.length) > MAX_TOPICS_COUNT)
+    return next(refused());
```

Comment:
The code will be cleaner if we extracted this logic into a separate variable.

 ```suggestion
  // ensure that user topics will not exceed the max num after insertion
  const exceededMaxAllowedTopics = (inDBTopicIds.length + addTopics.length - removeTopics.length) > MAX_TOPICS_COUNT
  if (exceededMaxAllowedTopics) return next(refused());
```